### PR TITLE
Update go.mod dependencies before release

### PR DESCRIPTION
**Pull request by juliusv**
Signed-off-by: Julius Volz <julius.volz@gmail.com>

### CODE REVIEW

File: cmd/promtool/main.go [hunk 1]
```diff
@@ -501,7 +501,8 @@ func QueryLabels(url *url.URL, name string, p printer) int {
 	// Run query against client.
 	api := v1.NewAPI(c)
 	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
-	val, err := api.LabelValues(ctx, name)
+	// TODO: Show warnings (second return value) somehow?
```
Comment (bwplotka) [inline cmd/promtool/main.go]: Can we add an issue for this? Otherwise, it will get lost.

Comment (juliusv) [reply cmd/promtool/main.go]: Created https://github.com/prometheus/prometheus/issues/5885 and added to comment here.

Now still have to figure out why a logger is `nil` in web tests...

**Comment by juliusv** [comment]
Reverted json-iterator version, hoping this will fix CI.

**Comment by juliusv** [comment]
I had push a number of followup fix commits to deal with fallout from various behavioral changes... please take another quick look.

**Pull request merged**